app/screens/settings/notification_settings_auto_responder/notification_settings_auto_responder.js

```diff
@@ -58,15 +58,13 @@ export default class NotificationSettingsAutoResponder extends PureComponent {
         };
     }
 
-    onNavigatorEvent = (event) => {
-        if (event.type === 'ScreenChangedEvent') {
-            switch (event.id) {
-            case 'willDisappear':
-                this.saveUserNotifyProps();
-                break;
-            }
-        }
-    };
+    componentDidMount() {
+        this.navigationEventListener = Navigation.events().bindComponent(this);
+    }
+
+    componentDidDisappear() {
+        this.saveUserNotifyProps();
```

Comment:
Can this be done in `componentWillUnmount`? I feel like this should actually be done when the settings themselves are changed, but that's a bigger change to this component.